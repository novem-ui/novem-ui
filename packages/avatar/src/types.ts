export type AvatarSizes = 'sm' | 'md' | 'lg'

export interface AvatarProps {
  description?: string
  image?: string
  name?: string
  size?: AvatarSizes
}
