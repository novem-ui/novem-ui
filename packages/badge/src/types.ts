import theme from '@novem-ui/theme'
import { Hierarchy } from '@novem-ui/base'
import { Theme } from '@emotion/react'

export type BadgeHierarchy = Exclude<Hierarchy, 'link'>
export type BadgeSize = 'sm' | 'lg'

export interface BadgeProps {
  hierarchy?: BadgeHierarchy
  size?: BadgeSize
  baseColor?: keyof Omit<Omit<typeof theme.colors.palette, 'base'>, 'darkGrey'>
}

export type BadgeUtilsFunction = (props: BadgeProps & { theme?: Theme }) => string
