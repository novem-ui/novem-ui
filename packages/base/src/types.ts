import { Property } from 'csstype'
import { BaseColor, Color, SpacingValue } from '@novem-ui/theme'
import { Theme } from '@emotion/react'

// Spacing types
export type SpacingPropName =
  | 'm'
  | 'mt'
  | 'mr'
  | 'mb'
  | 'ml'
  | 'mx'
  | 'my'
  | 'margin'
  | 'marginX'
  | 'marginY'
  | 'marginTop'
  | 'marginRight'
  | 'marginBottom'
  | 'marginLeft'
  | 'p'
  | 'pt'
  | 'pr'
  | 'pb'
  | 'pl'
  | 'px'
  | 'py'
  | 'padding'
  | 'paddingX'
  | 'paddingY'
  | 'paddingTop'
  | 'paddingRight'
  | 'paddingBottom'
  | 'paddingLeft'

export type SpacingPropValue = SpacingValue | string

export type SpacingPropEntry = [SpacingPropName, SpacingPropValue]

export type SpacingProps = { [key in SpacingPropName]?: SpacingPropValue }

export type CSSSpacingProps = Omit<
  SpacingProps,
  'm' | 'mt' | 'mr' | 'mb' | 'ml' | 'mx' | 'my' | 'p' | 'pt' | 'pr' | 'pb' | 'pl' | 'px' | 'py'
>

// Text types
export interface TypographyProps<As, Size> {
  align?: Property.TextAlign
  as?: As
  color?: Color | Property.Color
  decoration?: Property.TextDecoration
  direction?: Property.Direction
  family?: Property.FontFamily
  kerning?: Property.FontKerning
  letterSpacing?: Property.LetterSpacing
  lineHeight?: Property.LineHeight
  orientation?: Property.TextOrientation
  transform?: Property.TextTransform
  size?: Size | Property.FontSize
  shadow?: Property.TextShadow
  stretch?: Property.FontStretch
  fontStyle?: Property.FontStyle
  theme?: Theme
  variantCaps?: Property.FontVariantCaps
  variantNumeric?: Property.FontVariantNumeric
  weight?: Property.FontWeight
  whiteSpace?: Property.FontWeight
  wordBreak?: Property.WordBreak
  wordSpacing?: Property.WordSpacing
  writingMode?: Property.WritingMode
}

export type TextComponentType = 'heading' | 'text'
export type HeadingTags = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'

// Hierarchy
export type Hierarchy = 'loud' | 'quiet' | 'link'

// InputElement
export interface InputElementProps {
  theme?: Theme
  baseColor?: BaseColor
  error?: boolean | string | Error
  valid?: boolean
  disabled?: boolean
  readOnly?: boolean
}

// Layout

export interface LayoutProps {
  display?: Property.Display
  position?: Property.Position
  top?: Property.Top
  bottom?: Property.Bottom
  height?: Property.Height
  left?: Property.Left
  right?: Property.Right
  width?: Property.Width
}
