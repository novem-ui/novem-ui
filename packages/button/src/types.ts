import { Hierarchy, SpacingProps } from '@novem-ui/base'
import theme from '@novem-ui/theme'

export type ButtonSize = 'sm' | 'md' | 'lg'

export interface ButtonProps extends SpacingProps{
  hierarchy?: Hierarchy
  size?: ButtonSize
  baseColor?: keyof typeof theme.colors.palette
  disabled?: boolean
}
