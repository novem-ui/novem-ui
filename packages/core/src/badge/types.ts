import * as React from 'react'

export default interface IBadge {
  variant: string,
  children: React.ReactElement,
}
