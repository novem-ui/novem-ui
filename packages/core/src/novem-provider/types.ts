import { Theme } from "@novem-ui/theme";

export interface NovemProviderProps {
  theme?: Theme;
  reset?: boolean;
  baseColor?: string;
}
