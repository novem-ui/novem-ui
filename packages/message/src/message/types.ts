import theme from "@novem-ui/theme";

export interface ButtonActionTypes {
  text: string
  action: () => void
}

export interface MessageTypes {
  title?: string
  message: string
  onClose?: () => void
  button?: ButtonActionTypes
  baseColor?: keyof typeof theme.colors.palette
}
