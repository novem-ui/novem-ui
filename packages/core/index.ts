export { default as NovemProvider } from './src/novem-provider'

export { default as Badge } from '@novem-ui/badge'
export * from '@novem-ui/badge'

export { default as Button, ButtonProps } from '@novem-ui/button'

export { default as Switch, SwitchProps } from '@novem-ui/switch'

export { Heading, Label, Paragraph, HeadingProps, LabelProps, ParagraphProps } from '@novem-ui/text'

export { Input, TextField, TextFieldProps } from '@novem-ui/input'

export { default as theme } from '@novem-ui/theme'
export type { Theme } from '@novem-ui/theme'
