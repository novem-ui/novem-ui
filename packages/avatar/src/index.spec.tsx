import React from 'react'
import { render } from '@testing-library/react'
import { NovemProvider } from '@novem-ui/core'

import { AvatarProps } from './types'
import Avatar from '.'

const setup = (props: AvatarProps = {}) => {
  const result = render(
    <NovemProvider>
      <Avatar {...props} />
    </NovemProvider>
  )
  return result
}

describe('components/Avatar', () => {
  test('should match snapshot (default)', () => {
    const { asFragment } = setup({ image: 'a', name: 'Person Name' })

    expect(asFragment()).toMatchSnapshot()
  })

  test('should match snapshot (sm)', () => {
    const { asFragment } = setup({ image: 'a', name: 'Person Name', size: 'sm' })

    expect(asFragment()).toMatchSnapshot()
  })

  test('should match snapshot (lg)', () => {
    const { asFragment } = setup({ image: 'a', name: 'Person Name', size: 'lg' })

    expect(asFragment()).toMatchSnapshot()
  })

  test('should match snapshot (with description)', () => {
    const { asFragment } = setup({ image: 'a', name: 'Person Name', size: 'lg', description: 'Job description' })

    expect(asFragment()).toMatchSnapshot()
  })

  test('should match snapshot (no image)', () => {
    const { asFragment } = setup({ name: 'Person Name', size: 'lg' })

    expect(asFragment()).toMatchSnapshot()
  })

  test('should match snapshot (no name)', () => {
    const { asFragment } = setup({ image: 'a', size: 'lg' })

    expect(asFragment()).toMatchSnapshot()
  })
})
