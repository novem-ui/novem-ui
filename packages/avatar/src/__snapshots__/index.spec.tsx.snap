// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Avatar should match snapshot (default) 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  height: 3.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  box-sizing: border-box;
  border-radius: 2rem;
  background-color: #F1694B;
  height: 3.5rem;
  position: relative;
  width: 3.5rem;
  margin-right: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-4 {
  height: 100%;
  object-fit: cover;
  max-height: 100%;
  max-width: 100%;
  width: 100%;
}

.emotion-6 {
  box-sizing: border-box;
  height: 100%;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-8 {
  font-size: 14px;
  font-weight: 800;
  margin: 0;
}

<div
    class="emotion-0 emotion-1"
    height="14"
  >
    <div
      class="emotion-2 emotion-1"
      height="14"
      style="overflow: hidden;"
      width="14"
    >
      <img
        alt="Person Name's avatar image"
        class="emotion-4 emotion-5"
        src="a"
      />
    </div>
    <div
      class="emotion-6 emotion-1"
      direction="column"
      height="100%"
    >
      <p
        class="emotion-8 emotion-9"
      >
        Person Name
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`components/Avatar should match snapshot (lg) 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  height: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  box-sizing: border-box;
  border-radius: 2rem;
  background-color: #F1694B;
  height: 4rem;
  position: relative;
  width: 4rem;
  margin-right: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-4 {
  height: 100%;
  object-fit: cover;
  max-height: 100%;
  max-width: 100%;
  width: 100%;
}

.emotion-6 {
  box-sizing: border-box;
  height: 100%;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-8 {
  font-size: 16px;
  font-weight: 800;
  margin: 0;
}

<div
    class="emotion-0 emotion-1"
    height="16"
  >
    <div
      class="emotion-2 emotion-1"
      height="16"
      style="overflow: hidden;"
      width="16"
    >
      <img
        alt="Person Name's avatar image"
        class="emotion-4 emotion-5"
        src="a"
      />
    </div>
    <div
      class="emotion-6 emotion-1"
      direction="column"
      height="100%"
    >
      <p
        class="emotion-8 emotion-9"
      >
        Person Name
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`components/Avatar should match snapshot (no image) 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  height: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  box-sizing: border-box;
  border-radius: 2rem;
  background-color: #F1694B;
  height: 4rem;
  position: relative;
  width: 4rem;
  margin-right: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-4 {
  color: white;
  font-size: 24px;
  margin: 0;
  margin-bottom: 0.125rem;
}

.emotion-6 {
  box-sizing: border-box;
  height: 100%;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-8 {
  font-size: 16px;
  font-weight: 800;
  margin: 0;
}

<div
    class="emotion-0 emotion-1"
    height="16"
  >
    <div
      class="emotion-2 emotion-1"
      height="16"
      style="overflow: hidden;"
      width="16"
    >
      <p
        class="emotion-4 emotion-5"
        color="white"
      >
        PN
      </p>
    </div>
    <div
      class="emotion-6 emotion-1"
      direction="column"
      height="100%"
    >
      <p
        class="emotion-8 emotion-5"
      >
        Person Name
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`components/Avatar should match snapshot (no name) 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  height: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  box-sizing: border-box;
  border-radius: 2rem;
  background-color: #F1694B;
  height: 4rem;
  position: relative;
  width: 4rem;
  margin-right: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-4 {
  height: 100%;
  object-fit: cover;
  max-height: 100%;
  max-width: 100%;
  width: 100%;
}

<div
    class="emotion-0 emotion-1"
    height="16"
  >
    <div
      class="emotion-2 emotion-1"
      height="16"
      style="overflow: hidden;"
      width="16"
    >
      <img
        alt="undefined's avatar image"
        class="emotion-4 emotion-5"
        src="a"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`components/Avatar should match snapshot (sm) 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  height: 2.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  box-sizing: border-box;
  border-radius: 2rem;
  background-color: #F1694B;
  height: 2.5rem;
  position: relative;
  width: 2.5rem;
  margin-right: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-4 {
  height: 100%;
  object-fit: cover;
  max-height: 100%;
  max-width: 100%;
  width: 100%;
}

.emotion-6 {
  box-sizing: border-box;
  height: 100%;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-8 {
  font-size: 12px;
  font-weight: 800;
  margin: 0;
}

<div
    class="emotion-0 emotion-1"
    height="10"
  >
    <div
      class="emotion-2 emotion-1"
      height="10"
      style="overflow: hidden;"
      width="10"
    >
      <img
        alt="Person Name's avatar image"
        class="emotion-4 emotion-5"
        src="a"
      />
    </div>
    <div
      class="emotion-6 emotion-1"
      direction="column"
      height="100%"
    >
      <p
        class="emotion-8 emotion-9"
      >
        Person Name
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`components/Avatar should match snapshot (with description) 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  height: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  box-sizing: border-box;
  border-radius: 2rem;
  background-color: #F1694B;
  height: 4rem;
  position: relative;
  width: 4rem;
  margin-right: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-4 {
  height: 100%;
  object-fit: cover;
  max-height: 100%;
  max-width: 100%;
  width: 100%;
}

.emotion-6 {
  box-sizing: border-box;
  height: 100%;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-8 {
  font-size: 16px;
  font-weight: 800;
  margin: 0;
}

.emotion-10 {
  color: #9A9897;
  font-size: 14px;
  margin: 0;
}

<div
    class="emotion-0 emotion-1"
    height="16"
  >
    <div
      class="emotion-2 emotion-1"
      height="16"
      style="overflow: hidden;"
      width="16"
    >
      <img
        alt="Person Name's avatar image"
        class="emotion-4 emotion-5"
        src="a"
      />
    </div>
    <div
      class="emotion-6 emotion-1"
      direction="column"
      height="100%"
    >
      <p
        class="emotion-8 emotion-9"
      >
        Person Name
      </p>
      <p
        class="emotion-10 emotion-9"
        color="grey.500"
      >
        Job description
      </p>
    </div>
  </div>
</DocumentFragment>
`;
